src/Utils.cpp

```diff
@@ -31,15 +31,17 @@ std::optional<std::string> getAncestorPath(const std::string& path, const std::s
     // Append a "/" to the end of the parentPath to make searching easier
     auto parentPathWithSlash = *parentPath + "/";
 
+    // If the ancestor is the project root we need to rename it, applies only to non-DataModel projects
+    auto normalizedAncestorName = ancestorName == rootSourceNode->name ? "ProjectRoot" : ancestorName;
```

Comment:
Also, I think this check is applied too early.
Consider an example scenario of:
```
Foo
- Bar
   - Foo
      - Baz <- this calls FindFirstAncestor("Foo")
```
The returned path should be `ProjectRoot/Bar/Foo`, but this will return `ProjectRoot` since `normalizedAncestorName` will equal `ProjectRoot` rather than `Foo`.

Maybe this is resolved if we just do the check after the initial attempt?

Also, we should probably first add a check to see `startsWith(path, "ProjectRoot/")` so that we don't unknowingly apply this to DataModel projects (maybe that can be extracted out to a helper function "isDatamodelProject")